### PR TITLE
Solana 2 (for real this time)

### PR DESCRIPTION
**Pull request by kevinheavey**
Replacing #76 with this because that is stuck in rebase hell.

I had to do something a bit silly with `handle_sysvar` because the methods to write to individual fields haven't yet been restored to Solana 2.0. For now we just wipe the whole sysvar cache and rebuild it every time someone writes to a sysvar. Shouldn't matter too much.

I also reduced the NUM_GREETINGS const in the tests that use solana-program-test because it started breaking at higher numbers. That issue is entirely with solana-program-test so we don't care.

The first commit will be useful to refer to when Solana 2.1 comes out. Thanks to @OliverNChalk for doing some leg work there in #94 

Closes #94 
Closes #76

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -12,27 +12,30 @@ rust-version = "1.75.0"
...
-solana-program-runtime = "~1.18"
-solana-program-test = "~1.18"
-solana-sdk = "~1.18"
-solana-stake-program = "~1.18"
-solana-system-program = "~1.18"
-solana-vote-program = "~1.18"
-spl-associated-token-account = "3.0"
-spl-token = "3.5"
-spl-token-2022 = "3.0"
+solana-address-lookup-table-program = "2.0.13"
```
Comment (Aursen) [inline Cargo.toml:21]: Why will we don't use 2.0  instead of 2.0.13 here since solana is following semver right now ?

Comment (Aursen) [inline Cargo.toml:21]: Why will we don't use 2.0 instead of 2.0.13 here since solana is following semver right now ?

**Review by Aursen — CHANGES_REQUESTED** [review]

**Review by sonicfromnewyoke — COMMENTED** [review]
LGTM, have only 1 small remark regarding the dependencies

File: svm/Cargo.toml [hunk 2]
```diff
@@ -19,20 +19,24 @@ log.workspace = true
...
 solana-stake-program.workspace = true
+solana-svm.workspace = true
 solana-system-program.workspace = true
 solana-vote-program.workspace = true
 thiserror.workspace = true
 
 [dev-dependencies]
 criterion.workspace = true
 serde.workspace = true
+solana-compute-budget.workspace = true
```
[the remark this answers is not in the document]
Comment (kevinheavey) [reply svm/Cargo.toml]: agreed except for serde since it's optional in `[depdencies]`

File: Cargo.toml (hunk 1 above)
Comment (kevinheavey) [reply Cargo.toml]: They are only following semver for the crate listed here: https://docs.solanalabs.com/backwards-compatibility#rust-crates

This PR only works on Solana >= 2.05, < 2.1 (assuming they don't backport any breaking changes to 2.0.x)

I've relaxed the deps to `~2.0.5`

**Review by Aursen — APPROVED** [review]

**Pull request merged**